### ISSUE TITLE
Deep nested wlderchild inheritance is broken

### ISSUE THREAD

**Issue by tomlarkworthy** [firebase/blaze_compiler#48]
Test case added 

https://github.com/firebase/blaze_compiler/blob/master/test/codegen_test.ts#L271

**Comment by eygraber** [comment]
Is there an ETA on this? I've hit a wall because of it.

**Comment by tomlarkworthy** [comment]
I had a stab at it but it is quite tricky to fix. I am not even sure it's possible to fix. So I am really not sure. I am waiting for a flash or inspiration on how to tackle it.

**Comment by eygraber** [comment]
OK thanks. I'll try shuffling my schema around.

**Comment by eygraber** [comment]
Is this something that Bolt will be able to handle?

**Comment by tomlarkworthy** [comment]
Yeah bolt should not have the same problem, although Bolt's API is fairly
unstable ATM and not recommended for production yet.
